Sofa/framework/LinearAlgebra/src/sofa/linearalgebra/CompressedRowSparseMatrixConstraintEigenUtils.h

```diff
@@ -0,0 +1,248 @@
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <sofa/linearalgebra/CompressedRowSparseMatrixConstraint.h>
+#include <Eigen/Sparse>
+#include <sofa/type/Vec.h>
+//#include <sofa/defaulttype/RigidTypes.h>
+#include <cassert>
+#include <type_traits>
+#include <cstdlib>
+
+namespace sofa::linearalgebra
+{
+
+template< class TBlock >
+struct CompressedRowSparseMatrixToEigenSparse
+{
+
+};
+
+template <typename TVec>
+struct CompressedRowSparseMatrixToEigenSparseVec
+{
+    typedef typename TVec::Real Real;
+    typedef CompressedRowSparseMatrixConstraint< TVec > TCompressedRowSparseMatrix;
+    typedef Eigen::SparseMatrix<Real, Eigen::RowMajor> EigenSparseMatrix;
+
+
+    EigenSparseMatrix operator() (const TCompressedRowSparseMatrix& mat, std::size_t size)
+    {
+        std::size_t eigenMatSize = size * TVec::size();
+        EigenSparseMatrix eigenMat(eigenMatSize, eigenMatSize);
+
+        std::vector<Eigen::Triplet<Real> > triplets;
+
+        for (auto row = mat.begin(); row != mat.end(); ++row)
+        {
+            for (auto col = row.begin(), colend = row.end(); col !=colend; ++col)
+            {
+                const TVec& vec = col.val();
+                int   colIndex  = col.index() * TVec::size();
+
+                for (std::size_t i = 0; i < TVec::size(); ++i)
+                {
+                    triplets.emplace_back(Eigen::Triplet<Real>( row.index(), colIndex + i, vec[i]) );
+                }
+
+            }
+        }
+
+        eigenMat.setFromTriplets(triplets.begin(), triplets.end());;
+        eigenMat.makeCompressed();
```

Comment:
I don't think it is necessary to call this function as `setFromTriplets` outputs a compressed matrix: https://eigen.tuxfamily.org/dox/classEigen_1_1SparseMatrix.html#a8f09e3597f37aa8861599260af6a53e0
```suggestion
```